execution/engine/engine.go

```diff
@@ -164,11 +156,18 @@ func (ee *Engine) VerifyAndNotifyNewPayload(
 	// We abstract away some of the complexity and categorize status codes
 	// to make it easier to reason about.
 	switch {
-	// If we get accepted or syncing, we are going to optimistically
-	// say that the block is valid, this is utilized during syncing
-	// to allow the beacon-chain to continue processing blocks, while
-	// its execution client is fetching things over it's p2p layer.
+	case err == nil:
```

Comment:
just reordered the cases. I would rather have the `err == nil` as first case and use default for any error that does not require special handling.